test/e2e/tests/test_endpoint.py

```diff
@@ -0,0 +1,141 @@
+
+"""Integration tests for the EventBridge Endpoint API.
+"""
+
+import pytest
+import time
+import logging
+
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+from acktest.k8s import condition as condition
+from acktest import tags
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_eventbridge_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e.tests.helper import EventBridgeValidator
+
+RESOURCE_PLURAL = "endpoints"
+
+CREATE_WAIT_AFTER_SECONDS = 10
+UPDATE_WAIT_AFTER_SECONDS = 10
+DELETE_WAIT_AFTER_SECONDS = 10
+
+@pytest.fixture(scope="module")
+def event_bus():
+        resource_name = random_suffix_name("ack-test-bus", 24)
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["BUS_NAME"] = resource_name
+
+        # Load EventBus CR
+        resource_data = load_eventbridge_resource(
+            "eventbus",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, "eventbuses",
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        yield (ref, cr)
+
+        try:
+            _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+            assert deleted
+        except:
+            pass
+
+@pytest.fixture(scope="module")
+def endpoint(event_bus):
+        resource_name = random_suffix_name("ack-test-endpoint", 24)
+        _, eb_cr = event_bus
+
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["ENDPOINT_NAME"] = resource_name
+        replacements["EVENT_BUS_ARN"] = eb_cr["status"]["ackResourceMetadata"]["arn"]
+
+        # Load EventBus CR
+        resource_data = load_eventbridge_resource(
+            "endpoint",
+            additional_replacements=replacements,
+        )
+        logging.debug(resource_data)
+
+        # Create k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            resource_name, namespace="default",
+        )
+        k8s.create_custom_resource(ref, resource_data)
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        assert cr is not None
+        assert k8s.get_resource_exists(ref)
+
+        time.sleep(CREATE_WAIT_AFTER_SECONDS)
+
+        cr = k8s.wait_resource_consumed_by_controller(ref)
+
+        yield (ref, cr)
+
+        try:
+            _, deleted = k8s.delete_custom_resource(ref, 3, 10)
+            assert deleted
+        except:
+            pass
+
+
+@service_marker
+@pytest.mark.canary
+class TestEvenpoint:
```

Comment:
```suggestion
class TestEndpoint:
```

:)

Reply:
@jaypipes thx, this one is for the E2E master @A-Hilaly :)

Reply:
Sometimes, my dyslexia kicks off in very fashionable ways

Reply:
Surprised didn't drop an "eigenpoint"